kong/router/compat.lua

```diff
@@ -67,6 +75,68 @@ local LOGICAL_AND = atc.LOGICAL_AND
 local uuid_generator = assert(uuid.factory_v5('7f145bf9-0dce-4f91-98eb-debbce4b9f6b'))
 
 
+local function gen_for_nets(ip_field, port_field, vals)
+  if is_empty_field(vals) then
+    return nil
+  end
+
+  local nets_buf = buffer.new(64):put("(")
```

Comment:
We could make the buffer a parameter and pass from caller, so we could share the buffer through multiple calls

Reply:
`nets_buf ` could be a module level var, we can optimize it later.

Reply:
@chronolaw Aapo suggested that module-level could increase memory footprint. Request-level should be suitable here.

Reply:
done. thanks.